ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/ExpectancyCriterion.java

```diff
@@ -0,0 +1,73 @@
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.analysis.criteria;
+
+import org.ta4j.core.BarSeries;
+import org.ta4j.core.Position;
+import org.ta4j.core.TradingRecord;
+import org.ta4j.core.num.Num;
+
+/**
+ * Expectancy criterion (Kelly Criterion).
+ *
+ * Measures the positive or negative expectancy. The higher the positive number,
+ * the better a winning expectation.
+ *
+ * @see <a href=
+ *      "https://www.straightforex.com/advanced-forex-course/money-management/two-important-things-to-be-considered/</a>
+ *
+ */
+public class ExpectancyCriterion extends AbstractAnalysisCriterion {
+
+    private NumberOfWinningPositionsCriterion numberOfWinningPositionsCriterion = new NumberOfWinningPositionsCriterion();
+    private NumberOfLosingPositionsCriterion numberOfLosingPositionsCriterion = new NumberOfLosingPositionsCriterion();
+
+    @Override
+    public Num calculate(BarSeries series, TradingRecord tradingRecord) {
+        Num numberOfWinningPositions = numberOfWinningPositionsCriterion.calculate(series, tradingRecord);
+        Num numberOfLosingPositions = numberOfLosingPositionsCriterion.calculate(series, tradingRecord);
+        return calculate(series, numberOfWinningPositions, numberOfLosingPositions);
+    }
+
+    @Override
+    public Num calculate(BarSeries series, Position position) {
+        Num numberOfWinningPositions = numberOfWinningPositionsCriterion.calculate(series, position);
+        Num numberOfLosingPositions = numberOfLosingPositionsCriterion.calculate(series, position);
+        return calculate(series, numberOfWinningPositions, numberOfLosingPositions);
+    }
+
+    @Override
+    public boolean betterThan(Num criterionValue1, Num criterionValue2) {
+        return criterionValue1.isGreaterThan(criterionValue2);
+    }
+
+    private Num calculate(BarSeries series, Num numberOfWinningPositions, Num numberOfLosingPositions) {
+        Num one = series.numOf(1);
+        Num probabiltyToWinOnePosition = numberOfWinningPositions.dividedBy(series.numOf(100));
+        // Expectancy = (1 + AW/AL) * ProbabilityToWinOnePosition - 1
+        return (one.plus(numberOfWinningPositions.dividedBy(numberOfLosingPositions)))
```

Comment:
The site you are referring is not talking about number of trades but about amount of winning/loosing trades. E.g. (800/600) instead of 62/32 on https://www.straightforex.com/advanced-forex-course/money-management/two-important-things-to-be-considered

Reply:
Yes, I must use `AverageWinningPositionsCriterion` and `AverageLosingPositionsCriterion` (instead of numberOfPositions). **Those criteria are actually not existent in ta4j.** First step would be to merge https://github.com/ta4j/ta4j/pull/670. Then I add those criterions as a new PR, then I will correct this issue.

Reply:
Replaced by https://github.com/ta4j/ta4j/pull/704